polkadot/node/core/pvf/src/priority.rs

```diff
@@ -20,11 +20,12 @@ pub enum Priority {
 	/// Normal priority for things that do not require immediate response, but still need to be
 	/// done pretty quick.
 	///
-	/// Approvals and disputes fall into this category.
+	/// Backing falls into this category.
 	Normal,
 	/// This priority is used for requests that are required to be processed as soon as possible.
 	///
-	/// For example, backing is on a critical path and requires execution as soon as possible.
+	/// Disputes and approval are on a critical path and requires execution as soon as
+	/// possible to not delay finality.
```

Comment:
Backing was deemed more important than approvals and disputes? :fearful: 

If I understand it correctly that is an awesome find and fix. I was not aware of it. It also aligns much better with out backpressure plans - first thing you should stop doing when overworked is backing, NOT approvals. 

But despite the comments indicating that... I can't find any anything in the original code that sets anything to `Priority::Critical`. Huh.

Reply:
I am looking at the execution path as well, but that seem to require a lot more changes. 
However the backing execution backpressure to activate at the network level requires at least the backing group to be overloaded, but that also depends on the validator hardware, some might be faster or some might provision more CPUs.

Reply:
I've checked the original paritytech/polkadot#2710 and it seems it was like that from the very beginning :shrug: 
Maybe @pepyakin could remember why it was implemented like that

Reply:
Yeah, you might be right that this was far worse before async backing, but even now it is not so great either. Depending on the actual PVF, the preparation can take 10-20s so the relay parent could go out of scope. Given that we have 1 single worker doing this right now, finality can be delayed for quite a lot of time and this would affect finality on all parachains vs just one parachain that won't be able to get it's blocks included.

Reply:
Actually, we could also bump `prepare_workers_hard_max_num` from 1 to 2. This would be used only for preparing PVFs tagged with `Critical` priority.

Reply:
Sounds like a really good idea, but again, it would be good to have a number of CPU cores reserved in the hardware requirements for the maximum number of workers that may run in parallel.